addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/converters/DateTimeTypeWSDateValueConverter.java

```diff
@@ -44,10 +44,11 @@ public WSDateValue convertFromOHType(@NonNull DateTimeType from, @NonNull WSDate
         short year = (short) cal.get(Calendar.YEAR);
         byte month = (byte) (cal.get(Calendar.MONTH) + 1);
         byte day = (byte) cal.get(Calendar.DAY_OF_MONTH);
-        value.setYear(year);
-        value.setMonth(month);
-        value.setDay(day);
-        return value;
+        WSDateValue v = new WSDateValue(value);
```

Comment:
A fellow developer might use the class in the wrong way again. Why not making `WSDateValue` immutable as in removing all setters, getters and have final public fields only and various constructors to satisfy the required needs?